src/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,108 @@
+import Link from 'next/link';
+
+import config from '../../../package.json';
+import { Helmet } from 'react-helmet';
+
+import { GrPrevious as PreviousIcon, GrNext as NextIcon } from 'react-icons/gr';
+import { HiOutlineDotsHorizontal as Dots } from 'react-icons/hi';
+import styles from './Pagination.module.scss';
+
+const MAX_NUM_PAGES = 9;
+
+const { homepage = '' } = config;
+
+const Pagination = ({ pagesCount, currentPage, basePath }) => {
+  const path = `${basePath}/page/`;
+
+  const hasPreviousPage = pagesCount > 1 && currentPage > 1;
+  const hasNextPage = pagesCount > 1 && currentPage < pagesCount;
+
+  let hasPrevDots = false;
+  let hasNextDots = false;
+
+  function getPages() {
+    let pages = pagesCount;
+    let start = 0;
+    // If the number of pages exceeds the max
+    if (pagesCount > MAX_NUM_PAGES) {
+      // Set number of pages to the max
+      pages = MAX_NUM_PAGES;
+      const half = Math.ceil(MAX_NUM_PAGES / 2);
+      const isHead = currentPage <= half;
+      const isTail = currentPage > pagesCount - half;
+      hasNextDots = !isTail;
+      // If the current page is at the head, the start variable remains 0
+      if (!isHead) {
+        hasPrevDots = true;
+        // If the current page is at the tail, the start variable is set to
+        // the last chunk. Otherwise the start variable will place the current
+        // page at the middle
+        start = isTail ? pagesCount - MAX_NUM_PAGES : currentPage - half;
+      }
+    }
+    return [...new Array(pages)].map((_, i) => i + 1 + start);
+  }
+
+  const pages = getPages();
+
+  return (
+    <>
+      <Helmet>
+        {!hasPreviousPage && <link rel="canonical" href={`${homepage}${basePath}`} />}
+        {hasPreviousPage && <link rel="prev" href={`${homepage}${path}${currentPage - 1}`} />}
+        {hasNextPage && <link rel="next" href={`${homepage}${path}${currentPage + 1}`} />}
+      </Helmet>
+
+      <nav className={styles.nav} role="navigation" aria-label="Pagination Navigation">
+        {hasPreviousPage && (
+          <Link href={`${path}${currentPage - 1}`}>
+            <a className={styles.prev} aria-label="Goto Previous Page">
+              <PreviousIcon /> Previous
+            </a>
+          </Link>
+        )}
+
+        <ul className={styles.pages}>
+          {hasPrevDots && (
+            <li className={styles.dots}>
+              <Dots />
```

Comment:
That's a good point. I wasnt sure what to do here because as we already have Previous/Next links (with their aria-labels), it would end with two very similar labels on both sides (I was thinking on "More Posts"). Any ideas on what could be a proper aria-label here?  what do you think about using an `aria-hidden="true"` and leave it as visual-only info?

Reply:
yeah, i'd say either hidden or something like:
* "More pages not shown"
* "Navigation to pages 1-7 hidden"

I'm cool with whatever you think make sense

Reply:
Thanks! I think I would go with the "Navigation to pages 1-7 hidden" label, because I like that it gives the whole context of pages.